crates/rspack_plugin_copy/src/lib.rs

```diff
@@ -291,24 +294,50 @@ impl CopyRspackPlugin {
     let mut source = RawSource::from(source_vec.clone());
 
     if let Some(transform) = &pattern.transform {
+      logger.debug(format!(
+        "transforming content for '{}'...",
+        absolute_filename
+      ));
       match transform {
         Transformer::Fn(transformer) => {
-          let transformed = transformer(source_vec, absolute_filename.as_str()).await;
-          match transformed {
-            Ok(code) => {
-              source = code;
-            }
-            Err(e) => {
-              diagnostics
-                .lock()
-                .expect("failed to obtain lock of `diagnostics`")
-                .push(Diagnostic::error(
-                  "Run copy transform fn error".into(),
-                  e.to_string(),
-                ));
-            }
-          };
+          handle_transform(
+            transformer,
+            source_vec,
+            absolute_filename.clone(),
+            &mut source,
+            diagnostics,
+          )
+          .await
         }
+        Transformer::Opt((transformer, cache)) => match cache {
```

Comment:
I think we should all cache related logic if not support now

Reply:
it will not exposed to our users, cache is just a `None` here.

Reply:
i remove the logic here and add a comment here to wait cache support in the furture